### PR TITLE
Ghc 8.4 compatibility

### PR DESCRIPTION
**Pull request by angerman**
This adds the necessary Semigroup-Monoid-Proposal changes to haskell-chart.

### CODE REVIEW

File: chart-diagrams/Graphics/Rendering/Chart/Backend/Diagrams.hs [hunk 1]
```diff
@@ -126,7 +129,9 @@ cBackendToFile fo cb path = do
     EPS -> do
       let (d, a) = runBackend env cb
           opts = DEPS.PostscriptOptions path (D2.dims2D w h) DEPS.EPS
-      D.renderDia DEPS.Postscript opts d
+          eps = D.renderDia DEPS.Postscript opts d
+      withFile (opts D.^. DEPS.psfileName) WriteMode $ \h ->
```
Comment (timbod7) [inline chart-diagrams/Graphics/Rendering/Chart/Backend/Diagrams.hs]: Why is this addition required?

Comment (timbod7) [reply chart-diagrams/Graphics/Rendering/Chart/Backend/Diagrams.hs]: I was expecting the SMP changes (which look fine - thanks!)... but am unsure why this was also required...

Comment (angerman) [reply chart-diagrams/Graphics/Rendering/Chart/Backend/Diagrams.hs]: Ahh right. This is for https://github.com/diagrams/diagrams-postscript/commit/d64f48e9f7cf154799ada1fe056231c599d9e337 ... as diagrams-postscript SMP changes come after that commit. I guess haskell-chart is now dependent on a diagrams-postscript 1.5 release?

**Comment by timbod7** [comment]
Will the updated code build with both diagrams 1.4 and 1.5 ?

If not, I think I would prefer to keep compatibility with diagrams 1.4 until 1.5 is in stackage...

File: chart-diagrams/Graphics/Rendering/Chart/Backend/Diagrams.hs (hunk 1 above)
Comment (timbod7) [reply chart-diagrams/Graphics/Rendering/Chart/Backend/Diagrams.hs]: Yep - that looks like a good solution!

Comment (larskuhtz) [reply chart-diagrams/Graphics/Rendering/Chart/Backend/Diagrams.hs]: To me this change seems unrelated to making the package compile with GHC-8.4. By just adding `instance Semigroup Path where { (<>) = append }` the code in master compiled with `cabal install` and GHC-8.4.2.

Would it be possible to commit just the SMP changes and release a new version on Hackage?

Comment (angerman) [reply chart-diagrams/Graphics/Rendering/Chart/Backend/Diagrams.hs]: I'll try to split this into two tomorrow.
One that is SMP only, and affects only `chart` and another one that makes `chart-diagrams` SMP ready with a fix for the postscript 1.5 as detailed above.

**Comment by timbod7** [comment]
Hopefully resolved with: 3f58acada37a6508b63f0453895c7c9a6916ffcb
